### PR TITLE
Google Analytics Added

### PR DESCRIPTION
**Pull request by Bhavesh27**

### CODE REVIEW

**Comment by HarshdipD** [comment]
@Bhavesh27 Can you please explain how to make the analytics work on the dev and prod environment in this PR thread? 
I have my analytic ID added, and I want to replicate and see live data like button click number before I can push this to main. Screenshots would help! Thanks again!

File: popup.js
Comment (HarshdipD) [inline popup.js:110]: I believe this code doesn't actually send the event click to the analytics - the official docs use this as a function that is sent as an addEventListener. Let me know if I am misunderstanding this. I cannot see the button click box on the overview page on the Analytics page either.

File: key.js
Comment (HarshdipD) [inline key.js:2]: Can you see if you can use both these variables as `const` instead? I should have seen this before, since they don't change const will be a better alternative.

**Review by HarshdipD — CHANGES_REQUESTED** [review]
Good job and this would be of great help! Can you please fix the requested changes and add page tracking for the settings page too? I'd love to see how many users are visiting it and how many authorize it with Trello, or bounce away.

File: popup.js
Comment (Bhavesh27) [reply popup.js:110]: @HarshdipD  Actually According to docs we just have to push the event to gaq global array. Analytics script listens to gaq array changes and fires the event asynchronously in background.

File: key.js
Comment (Bhavesh27) [reply key.js]: Done

File: popup.js
Comment (HarshdipD) [reply popup.js:110]: Thanks, Bhavesh, just to confirm if I'm getting this right, I don't see the button name on the analytics overview page, as it is supposed to be in docs. Could you provide a screenshot of it working for you so I know where to look?

**Comment by HarshdipD** [comment]
@Bhavesh27  just waiting on one comment before I merge this!

**Comment by Bhavesh27** [comment]
@HarshdipD Sorry. I got occupied. Actually In the google sample example we have Multiple Buttons thats why we are passing second argument as **event.target.id**. So, as to recognize which button is clicked. On our Project, we only have one button, **Add To Trello** and thats what i have put up as second arguement of the GA call to recognize that button is clicked.
PR is Good to Go from my end.

**Comment by HarshdipD** [comment]
@Bhavesh27  No worries! I got what you mean, that works! Also, in my previous review, I mentioned if you can track the settings page and Authorize button too, as it falls under this same issue. 
Additionally, I think the code might have merge conflicts now because the main branch has been updated a couple of time, could you please make sure it's fine from your end? Thank you!

**Comment by Bhavesh27** [comment]
> @Bhavesh27 No worries! I got what you mean, that works! Also, in my previous review, I mentioned if you can track the settings page and Authorize button too, as it falls under this same issue.
> Additionally, I think the code might have merge conflicts now because the main branch has been updated a couple of time, could you please make sure it's fine from your end? Thank you!

@HarshdipD Sure i will resolve the queries. On settings Page, Popup.js is included so we are already firing the pageview event for Settings Page. Event for Authorize Button can be added.

My doubt is if we want Settings Page to be Different i can extract the analytics code from Popup.js and create a separate .js file that can be added to both popup.html and settings.html and we can remove popup.js from the settings page.

**Comment by HarshdipD** [comment]
@Bhavesh27  This works perfectly! Thank you so much for this PR, and I'm sure these analytics will help the tool become better by tracking user activities! 
I'll merge this PR now. Thanks again! Feel free to take any other issue if you're interested and star this repo!

**Pull request merged**
